lib/puppet/provider/oneview_server_profile/c7000.rb

```diff
@@ -24,14 +24,17 @@
   mk_resource_methods
 
   def exists?
-    @data = data_parse
-    empty_data_check([:found, :get_available_targets, :get_available_networks, :get_available_servers, :get_compliance_preview,
-                      :get_messages, :get_profile_ports, :get_transformation, :absent])
+    super([:found, :get_available_targets, :get_available_networks, :get_available_servers,
+           :get_compliance_preview, :get_messages, :get_profile_ports, :get_transformation, :absent])
+
     # gets the connections' uris
     connections_parse if @data['connections']
     # gets the hash of filters for queries; in case it does not exist, query will be nil
     @query = @data.delete('query_parameters')
-    !@resourcetype.find_by(@client, @data).empty?
+
+    return @resourcetype.find_by(@client, @data).any? unless resource['ensure'] == :present
```

Comment:
Due to the PR #102 I'm kind of questioning the relevance of this PR now.
What exactly is being done here that was not before? If the state is present it is currently applying the update and returning true, which is basically what would be done anyway on the 'create' method.
As it is, I don't think this is adding much/any value. Please lemme know if I'm overlooking something.

Also, definitely a must now, more than before, to have the scenarios for which the server profile fails to perform idempotently and against which this PR was ran added to the PR and/or to the issue related.